### DISCUSSION TITLE
There are some problems with running the results after incorporating the math formula

### DISCUSSION THREAD

**Discussion by SCZ1174** [Open-Systems-Pharmacology/Forum#1649] [Questions & Problems] [answered]
Hello, everyone,
I'm creating math equations in mobi, but the results are a little different from what I envisioned, I'll briefly describe my problem with an example.

I added a formula with the following parameters:
A: Dimension: Fraction, Value: 6.3
B: Dimension: Time, Value: 18.48 h
C: Dimension: Dimensionless, Formula: A×cos((2×pi/24)×(Time-B))
![d0bc726cd4330aeabb1f3a2b8dd6e19](https://github.com/Open-Systems-Pharmacology/Forum/assets/86552551/941526d8-880d-4e8a-9186-2ded21bfc4eb)
![ca3e565cded700d6179ef2956b53679](https://github.com/Open-Systems-Pharmacology/Forum/assets/86552551/a9835815-41f1-47a5-b692-0791c72eb2d4)

I believe that when I set it up this way, "C" is only related to "A", "B" and "Time", and any other parameters should not affect "C".
Plot C, The result is plotted below: 
![e35f134302229fa4fe89dcae599f124](https://github.com/Open-Systems-Pharmacology/Forum/assets/86552551/f32e3461-68a9-4a1a-83f3-b3ba894d2a4a)

The results were not what I expected, and I even thought they were "wrong".
I used Excel to generate some values and I went to graphing and the results are as follows:
![13d3691322f761ec3a5520c1c26fe57](https://github.com/Open-Systems-Pharmacology/Forum/assets/86552551/3a87df0e-37c2-4dad-a4fb-a11d58be6315)

This result should be what I want.
I don't know why I get this result, is it due to defining a different "Dimension"? Do we need to define the unit before using "Time" as a variable? Or do I need to define all "Dimensions" as "Dimensionless"?

**Comment by Yuri05** [comment, marked as answer]
You must be aware that all entities are converted to the **base OSP units** for calculations.
Base unit for the time dimension is **[min]**. So your formulas evaluates to
`A×cos((2×pi/24)×(Time_in_Minutes-18.48*60))`

**Reply by SCZ1174** [reply]
Thank you very much for your answer! This problem has been bugging me all day and now it's finally solved. hhh :)
